pandas/tests/base/test_unique.py

```diff
@@ -12,7 +15,11 @@
 def test_unique(index_or_series_obj):
     obj = index_or_series_obj
     obj = np.repeat(obj, range(1, len(obj) + 1))
-    result = obj.unique()
+    with tm.maybe_produces_warning(
+        PerformanceWarning,
+        pa_version_under2p0 and str(index_or_series_obj.dtype) == "string[pyarrow]",
```

Comment:
It appears that with a pyarrow backed StringSrray, we are only testing Index here, not Series?

Also don't need the `str` cast, dtype equality to the string form should work.

Reply:
> It appears that with a pyarrow backed StringSrray, we are only testing Index here, not Series?

Looks so based on the fixture in `pandas/conftest.py`

```
if has_pyarrow:
    idx = Index(pd.array(tm.makeStringIndex(100), dtype="string[pyarrow]"))
    indices_dict["string-pyarrow"] = idx
```

Fixed the comparison